include/kernel.h

```diff
@@ -1769,6 +1769,25 @@ static inline int64_t k_uptime_get(void)
 	return k_ticks_to_ms_floor64(k_uptime_ticks());
 }
 
+/**
+ * @brief Get system uptime, in microseconds.
+ *
+ * This routine returns the elapsed time since the system booted,
+ * in microseconds.
+ *
+ * @note
+ *    While this function returns time in microseconds, it does
+ *    not mean it has microsecond resolution. The actual resolution depends on
+ *    @option{CONFIG_SYS_CLOCK_TICKS_PER_SEC} config option.
+ *
+ * @return Current uptime in microseconds.
+ */
+static inline int64_t k_uptime_get_us(void)
```

Comment:
My preference would not be to have a separate API variant for every possible unit and to just have ticks be the main API (with the ms API present for backwards compatibility).  But if we're going to do it this we we should probably do all the known units at once instead of adding them piecemeal like this.